test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/XPlatConfigTests.cs

```diff
@@ -217,6 +219,119 @@ public void ConfigGetConfigKeyCommand_HelpMessage_Success()
             DotnetCliUtil.VerifyResultSuccess(result, helpMessage);
         }
 
+        [Theory]
+        [InlineData("signatureValidationMode", "accept")]
+        [InlineData("maxHttpRequestsPerSource", "64")]
+        public void ConfigSetCommand_AddNewConfigSettingWithConfigFileArg_Success(string key, string value)
+        {
+            // Arrange & Act
+            using var testInfo = new TestInfo("NuGet.Config");
+            var filePath = Path.Combine(testInfo.WorkingPath, "NuGet.Config");
+
+            var result = CommandRunner.Run(
+                DotnetCli,
+                Directory.GetCurrentDirectory(),
+                $"{XplatDll} config set {key} {value} --configfile {filePath}",
+                waitForExit: true);
+
+            var settings = Configuration.Settings.LoadDefaultSettings(
+                testInfo.WorkingPath,
+                configFileName: filePath,
+                machineWideSettings: new XPlatMachineWideSetting());
+
+            var configSection = settings.GetSection("config");
+            var values = configSection?.Items.Select(c => c as AddItem).Where(c => c != null).ToList();
+            var configItems = values.Where(i => i.Key == key);
```

Comment:
My comment above suggests that we should be explicit if configSection is expected to be null or not in this test. If we create a test where configSection is null, note that this would result in a NullReferenceException because `values` would be null.